packages/tools/src/eventListeners/touch/touchStartListener.ts

```diff
@@ -290,126 +333,128 @@ function _onTouchDrag(evt: TouchEvent) {
-  document.removeEventListener('touchend', _onTouchEnd);
-}
-
-function _onTouchTap(evt: TouchEvent): void {
   const currentPointsList = getTouchEventPoints(evt, state.element);
   const lastPointsList = _updateTouchEventsLastPoints(
     state.element,
     state.lastPointsList
   );
+  const deltaPoints =
+    currentPointsList.length === lastPointsList.length
+      ? getDeltaPoints(currentPointsList, lastPointsList)
+      : getDeltaPoints(currentPointsList, currentPointsList);
+  const deltaDistance =
+    currentPointsList.length === lastPointsList.length
+      ? getDeltaDistanceBetweenIPoints(currentPointsList, lastPointsList)
+      : getDeltaDistanceBetweenIPoints(currentPointsList, currentPointsList);
+  const eventDetail: EventTypes.TouchEndEventDetail = {
+    event: evt,
+    eventName: TOUCH_END,
+    element: state.element,
+    renderingEngineId: state.renderingEngineId,
+    viewportId: state.viewportId,
+    camera: {},
+    startPointsList: copyPointsList(state.startPointsList),
+    lastPointsList: copyPointsList(lastPointsList),
+    currentPointsList,
+    startPoints: getMeanTouchPoints(state.startPointsList),
+    lastPoints: getMeanTouchPoints(lastPointsList),
+    currentPoints: getMeanTouchPoints(currentPointsList),
+    deltaPoints,
+    deltaDistance,
+  };
+
+  triggerEventCallback(eventDetail.element, TOUCH_END, eventDetail);
+  _checkTouchTap(evt);
 
+  // reset to default state
+  state = JSON.parse(JSON.stringify(defaultState));
+  document.removeEventListener('touchmove', _onTouchDrag);
+  document.removeEventListener('touchend', _onTouchEnd);
+}
+
+function _checkTouchTap(evt: TouchEvent): void {
+  const currentTime = new Date().getTime();
+  const startTime = state.startTime.getTime();
+  if (currentTime - startTime >= tapState.tapWindow) return;
+
+  // first tap, initialize the state
+  if (tapState.taps === 0) {
+    tapState.element = state.element;
+    tapState.renderingEngineId = state.renderingEngineId;
+    tapState.viewportId = state.viewportId;
+    tapState.startPointsList = state.startPointsList;
+    tapState.startTime = new Date();
+  }
+
+  // check that taps are within interval
+  if (currentTime - tapState.startTime.getTime() >= tapState.tapWindow) return;
+
+  // subsequent tap is on a different element
   if (
-    state.taps > 0 &&
-    getDeltaDistance(currentPointsList, lastPointsList).canvas <
-      state.tapMaxDistance
+    tapState.taps > 0 &&
+    !(
+      tapState.element == state.element &&
+      tapState.renderingEngineId == state.renderingEngineId &&
+      tapState.viewportId == state.viewportId
+    )
   ) {
-    state.taps = state.taps + 1;
+    return;
   }
 
-  if (state.taps === 0) {
-    state.taps = state.taps + 1;
-  }
+  const currentPointsList = getTouchEventPoints(evt, tapState.element);
+  const distanceFromStart = getDeltaDistance(
+    currentPointsList,
+    tapState.startPointsList
+  ).canvas;
+
+  if (distanceFromStart >= tapState.tapMaxDistance) return;
```

Comment:
48px is about the max distance that's reasonable for index finger tapping from light experimentation. We can change it to something lower if needed. A future PR might make these settings configurable.

Related to the clamping (-15, 10), the reason it was asymmetric was because left and right handedness. These value adjustments are probably better left for config.